front: improve the new scenario header

kmer2016 opened the pull request:
This PR is the continuation of the work started in [this preview PR](https://github.com/OpenRailAssociation/osrd/pull/11627) on the header of the new scenario view.

It fully implements the header according to the specifications detailed in [this mockup](https://www.sketch.com/s/008d6980-d719-4bdc-bb2a-b4c2dda0dc1e/f/69619A0B-E473-4AB5-81BF-97BC06FB02FE#Inspect).

You can [browse the commits](https://github.com/OpenRailAssociation/osrd/pull/11658/commits) to follow the step-by-step implementation process.

To test the behavior with long labels, you can manually set a very long string as a user name or label (e.g., "_super_long_name_for_testing_purposes") and verify that ellipsis is applied to prevent collisions. Hovering over the truncated label should reveal its full content and temporarily hide the board display management component.

kmer2016 committed:
front: implement scenario info section specifications.

The name and information component is justified left and made of four distinct zones:
- Active area for the close button
- Inactive area between the close button and the scenario informations access
- Active area to access scenario informations; expands with the name character count
- A fixed width spacer preventing collision with the board display management component.

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: implement the hover over the name and informations section.

- When hovering the close button, the name fades out and the “close” label fades in.

- When hovering the scenrio’s name, the label and the disclosure icon change colour.

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: implement spec. for the board

The board display management area is always horizontally in the viewport. It’s made of:
 - A fixed width spacer preventing collision with the scenario’s name and information access
 - A list of family toggle buttons
 - Inactive areas between these buttons
 - A fixed width spacer preventing collision with the user’s name and help access

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: implement board toggle states

Implement visual states for board toggle buttons according to specs:
- Add ON state with rest/hover/active variations
- Add OFF state with rest/hover/active variations

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: refactor ScenarioHeader layout and styles for improved structure

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: make scenario name and user name ellipse when viewport width reduced

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: show full text labels on hover when truncated

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: move infra-loading styles to dedicated file

Move .infra-loaded class from scenario.scss to _infraLoadingState.scss for better modularity and reusability.

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: implement scenario details panel

Add collapsible details panel for scenario information that:
- Displays below the header when expanded
- Shows scenario description, electrical profile and infra state
- Maintains edit functionality via modal

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: open study page when close the current scenario

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: specify a color when scenario name container is active

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -19,12 +19,19 @@
 
+    .spacer {
+      width: 8px;
+      height: 32px;
```
You don't need to specify the height I think (tested and it works on my laptop)
```suggestion
```

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -90,14 +91,34 @@
+
+    .spacer {
+      width: 32px;
+      height: 100%;
```
```suggestion
```

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -90,14 +91,34 @@
+      .board-separator {
+        width: 8px;
+        height: 100%;
+        background-color: var(--ambientB20);
```
You don't need these 2 fields
```suggestion
```

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -12,9 +13,21 @@ type ScenarioHeaderProps = {
+    });
+  };
+
+  console.log({ activeBoards });
```
To remove before merging

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -107,11 +107,44 @@
+        }
+
+        // Off state
+        &.off {
```
Why not use `not(.on)` ?

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -107,11 +107,44 @@
+          &:active {
+            color: var(--grey50);
+            background-color: var(--ambientB15);
+          }
```
You have inverted the active and hover style (mockup order: on rest, on hover, on active):
![image](https://github.com/user-attachments/assets/f512d481-299f-404a-85aa-69476a4b5c8f)

Same for the off state

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -156,20 +156,32 @@
+
+    .spacer {
+      width: 32px;
+      height: 100%;
```
```suggestion
```

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -156,20 +156,32 @@
-    .help-icon {
+    .inactive-area {
+      width: 8px;
+      height: 100%;
```
```suggestion
```

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -80,10 +80,15 @@ const ScenarioHeader = ({ scenarioName }: ScenarioHeaderProps) => {
+
+        <div className="inactive-area" />
+
+        <button className="help-button" type="button">
```
```suggestion
```
No need to add this

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -80,10 +80,15 @@ const ScenarioHeader = ({ scenarioName }: ScenarioHeaderProps) => {
-        <button className="help-icon" type="button">
+        </button>
+
+        <div className="inactive-area" />
```
Hum why not use margin actually ? instead of these inactive area ?

kmer2016 replied:
I initially considered using margin, but according to the design specs, the spacing between the child elements should be inactive — meaning it should not allow any user interaction.
Using margin alone wouldn’t prevent interaction in those areas, whereas inserting spacer `<div>` makes it possible to explicitly block pointer events as required.

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -8,19 +8,32 @@
+  .spacer {
+    width: 32px;
+    height: 100%;
+    border: 1px solid red;
```
```suggestion
```

clarani reviewed (commented):
Some bugs:
- if I put my mouse at the right side of the name of the scenario part (but not on it), the scenario name is highlighted
![image](https://github.com/user-attachments/assets/cd3f5abc-2397-4b14-8e96-349cd1b10d55)

- if I hover the chevron, the scenario name is not fully displayed
![image](https://github.com/user-attachments/assets/61aea1a6-834a-411a-a849-aed99ec82640)

- The cross and the text on its right are on the same line
![image](https://github.com/user-attachments/assets/0b2ea78b-7fe5-45b3-8959-3294f1a42248)

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -26,7 +33,28 @@ const ScenarioHeader = ({ scenarioName }: ScenarioHeaderProps) => {
   };
 
-  console.log({ activeBoards });
+  const checkTruncation = useCallback(() => {
```
This useCallback is not reset since its array of dependances is empty. You can simply remove the callback and declare the function in the useEffect directly.

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -26,7 +33,28 @@ const ScenarioHeader = ({ scenarioName }: ScenarioHeaderProps) => {
+        ...prev,
+        username: usernameRef.current!.scrollWidth > usernameRef.current!.clientWidth,
+      }));
+    }
```
To avoid extra setState and re-renders
```suggestion
    setIsTruncated((prev) => ({
      scenarioName: scenarioNameRef.current
        ? scenarioNameRef.current.scrollWidth > scenarioNameRef.current.clientWidth
        : prev.scenarioName,
      username: usernameRef.current
        ? usernameRef.current.scrollWidth > usernameRef.current.clientWidth
        : prev.username,
    }));
```

clarani commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -49,7 +49,7 @@
 
         & ~ .close-label {
           text-transform: capitalize;
-          opacity: 1;
```
You should keep both the opacity props and the display props, otherwise you have this weird effect (the scenario's name seems to blink)
[Capture vidéo du 2025-05-22 12-04-35.webm](https://github.com/user-attachments/assets/f028fccb-aa32-418c-b9ce-219e8f570307)

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -1,29 +1,47 @@
-  const { t } = useTranslation('common');
+  const { openModal } = useModal();
+
+  const { t } = useTranslation(['translation', 'common', 'operational-studies']);
```
You have more operational-studies translations in this file, so it would be better to give this simple arg:
```suggestion
  const { t } = useTranslation('operational-studies');
```

Then you can remove all the `t(operational-studies:...)` => `t(...)`

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -1,29 +1,47 @@
 
   const boards = ['Traits', 'Map', 'Macro', 'STD', 'SDD', 'Table', 'Conflicts'];
 
+  const { data: electricalProfileSets } =
+    osrdEditoastApi.endpoints.getElectricalProfileSet.useQuery();
+
+  const electricalProfileSet = useMemo(
+    () =>
+      electricalProfileSets?.find((profile) => profile.id === scenario.electrical_profile_set_id),
+    [electricalProfileSets]
+  );
```
nit: since electricalProfileSets is not used anywhere else in the application, you can use RTK selectFromResults:

```suggestion
  const { electricalProfileSet } = osrdEditoastApi.endpoints.getElectricalProfileSet.useQuery(
    undefined,
    {
      selectFromResult: (response) => ({
        ...response,
        electricalProfileSet: response.data?.find(
          (profile) => profile.id === scenario.electrical_profile_set_id
        ),
      }),
    }
  );
```

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -1,29 +1,47 @@
     scenarioName: false,
     username: false,
   });
+  const [isScenarioDetailsVisible, setIsScenarioDetailsVisible] = useState(false);
```
typo: `areScenarioDetailsVisible` or `isScenarioDescriptionVisible` ?

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -57,83 +79,135 @@ const ScenarioHeader = ({ scenarioName }: ScenarioHeaderProps) => {
-        >
-          {username}
-        </button>
+        {/* scenario details */}
```
This line should be after the closure balise `</div>`

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
The commit `front: make the scenario name container clickable to display details ` shoud be merged into the previous ones

kmer2016 committed:
front: implement user info section

- Fixed width spacer to prevent collision with boards
- Username button for accessing settings
- Inactive area between username and help
- Help button with icon

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: fix hover scenario-name-label

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: fix scenario name blinking

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

kmer2016 committed:
front: simplify translation keys on ScenarioHeader

Signed-off-by: nncluzu <ngamenichaka@yahoo.fr>

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -1,49 +1,210 @@
+  const { openModal } = useModal();
+  const navigate = useNavigate();
+
+  const { t } = useTranslation(['operational-studies']);
```
```suggestion
  const { t } = useTranslation('operational-studies');
```

clarani commented on front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx:
```diff
@@ -1,49 +1,210 @@
+                >
+                  {board}
+                </button>
+                {index < boards.length - 1 && <div className="inactive-area" />}
+              </Fragment>
+            ))}
```
~Can you do that ? I think you can even remove the fragment if you do that~
```suggestion
              </Fragment>
            ))}
            {index < boards.length - 1 && <div className="inactive-area" />}
            
```

clarani replied:
Not relevant, sorry !

thibautsailly commented:
Here are some color modifications to do based on what Clara showed me earlier today:

```
On/Rest : grey80/ambientB5
On/Hover : black100/white100
On/Active : grey20/ambientB15

Off/Rest : grey30/ambientB15
Off/Hover : grey30/white100
Off/Active : grey50/ambientB5
```

Thank you!

clarani reviewed (approved):
LGTM & tested :white_check_mark:
